javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,38 @@
+const promise1 = moveElement(document.getElementById("red-button"), {
+  x: 20,
+  y: 300,
+});
+const promise2 = moveElement(document.getElementById("blue-button"), {
+  x: 400,
+  y: 300,
+});
+const promise3 = moveElement(document.getElementById("green-button"), {
+  x: 400,
+  y: 20,
+});
```

Comment:
Unfortunately in `translateOneByOne()` the elements are not moving one by one but they are moving at the same time. This is because the `promise1`, `promise2` and `promise3` are declared in the start of the program so they are executed instantly. 

To fix it, you can declare them as arrow functions and call them inside `Promise.all()`. This way they will only be executed when they are called. 

For example:



```suggestion
      const promise1 = ()=> moveElement(document.getElementById("red-button"), {
        x: 20,
        y: 300,
      });
      const promise2 = ()=> moveElement(document.getElementById("blue-button"), {
        x: 400,
        y: 300,
      });
      const promise3 = ()=> moveElement(document.getElementById("green-button"), {
        x: 400,
        y: 20,
      });

...

function translateAllAtOnce() {
  Promise.all([promise1(), promise2(), promise3()]).then(() =>
    console.log("all circles moved")
  );
}
translateAllAtOnce();
```

Reply:
So thanks with this comments. I was stuck with it.